src/ucp/core/ucp_request.inl

```diff
@@ -355,7 +376,7 @@ ucp_request_send_state_init(ucp_request_t *req, ucp_datatype_t datatype,
     VALGRIND_MAKE_MEM_UNDEFINED(&req->send.state.dt.offset,
                                 sizeof(req->send.state.dt.offset));
 
-    req->send.state.uct_comp.func = NULL;
+    ucp_request_send_state_clear(req, NULL);
```

Comment:
using this function in many places is adding too much extra overhead
just need to initialize missing fields where needed in few places

Reply:
right, added it only in required places only (where these values are not initialized)

Reply:
do we really need to reset all fields in these locations or only some of them?
also, if uct_comp.func==NULL, maybe we don't have to reset count,status?

Reply:
I added initialization based on vagrind failures in CI
>if uct_comp.func==NULL, maybe we don't have to reset count,status?

ok, updated evaluation